spec/rspec/core/configuration_spec.rb

```diff
@@ -1229,20 +1229,124 @@ def metadata_hash(*args)
+            output = StringIO.new
+            config.output_stream = output
+
+            allow(config.output_stream).to receive_messages :tty? => true
+
+            expect(config.color_enabled?).to be_truthy
+            expect(config.color_enabled?(output)).to be_truthy
+          end
+        end
+
+        context "with !output.tty?" do
+          it "sets color_enabled?" do
+            output = StringIO.new
+            config.output_stream = output
+
+            allow(config.output_stream).to receive_messages :tty? => false
+
+            expect(config.color_enabled?).to be_truthy
+            expect(config.color_enabled?(output)).to be_truthy
+          end
+        end
+      end
+
+      context "#force_color=false, #no_color=true" do
+        before { config.no_color = true }
+
+        context "with output.tty?" do
+          it "sets !color_enabled?" do
+            output = StringIO.new
+            config.output_stream = output
+
+            allow(config.output_stream).to receive_messages :tty? => true
+
+            expect(config.color_enabled?).to be_falsy
+            expect(config.color_enabled?(output)).to be_falsy
+          end
+        end
+
+        context "with !output.tty?" do
+          it "sets !color_enabled?" do
+            output = StringIO.new
+            config.output_stream = output
+
+            allow(config.output_stream).to receive_messages :tty? => false
+
+            expect(config.color_enabled?).to be_falsy
+            expect(config.color_enabled?(output)).to be_falsy
+          end
+        end
+      end
+
+      describe "#force_color=" do
+        it "prefers incoming cli_args" do
+          config.output_stream = StringIO.new
+          allow(config.output_stream).to receive_messages :tty? => true
+          config.force :force_color => true
+          config.force_color = false
+          expect(config.force_color).to be_truthy
+        end
+      end
+
+      describe "#no_color=" do
+        it "prefers incoming cli_args" do
+          config.output_stream = StringIO.new
+          allow(config.output_stream).to receive_messages :tty? => true
+          config.force :no_color => true
+          config.no_color = false
+          expect(config.no_color).to be_truthy
+        end
+      end
+    end
+
     describe "#color=" do
       context "given false" do
         before { config.color = false }
 
         context "with config.tty? and output.tty?" do
-          it "does not set color_enabled?" do
+          it "sets color_enabled?" do
             output = StringIO.new
             config.output_stream = output
 
             config.tty = true
             allow(config.output_stream).to receive_messages :tty? => true
 
-            expect(config.color_enabled?).to be_falsy
-            expect(config.color_enabled?(output)).to be_falsy
+            expect(config.color_enabled?).to be_truthy
+            expect(config.color_enabled?(output)).to be_truthy
```

Comment:
On the surface, it seems a little unfortunate to change the behavior here.  In theory, it would be more backwards compatible to make explicit calls to `config.color = false` behave the same as before and only do the deafulting described here if `config.color` is unset (e.g. `nil`).  In practice, I'm not sure that it's worth the effort.

Thoughts?